src/main/java/com/lothrazar/cyclic/item/endereye/ItemEnderEyeReuse.java

```diff
@@ -2,54 +2,99 @@
 
 import com.lothrazar.cyclic.base.ItemBase;
 import com.lothrazar.cyclic.util.UtilItemStack;
+import com.lothrazar.cyclic.util.UtilWorld;
 import net.minecraft.advancements.CriteriaTriggers;
 import net.minecraft.entity.player.PlayerEntity;
 import net.minecraft.entity.player.ServerPlayerEntity;
 import net.minecraft.item.ItemStack;
 import net.minecraft.stats.Stats;
-import net.minecraft.util.ActionResult;
-import net.minecraft.util.Hand;
-import net.minecraft.util.SoundCategory;
-import net.minecraft.util.SoundEvents;
+import net.minecraft.util.*;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 import net.minecraft.world.gen.ChunkGenerator;
+import net.minecraft.world.gen.feature.IFeatureConfig;
+import net.minecraft.world.gen.feature.StructureFeature;
 import net.minecraft.world.gen.feature.structure.Structure;
 import net.minecraft.world.server.ServerWorld;
+import net.minecraftforge.fml.ModList;
+import net.minecraftforge.fml.RegistryObject;
+import net.minecraftforge.registries.ForgeRegistries;
+
+import javax.annotation.Nullable;
 
 public class ItemEnderEyeReuse extends ItemBase {
 
   public ItemEnderEyeReuse(Properties properties) {
     super(properties.maxDamage(256));
   }
 
+  //compat with Repurposed Structures Mod see #1517
+  private static final String RS_MODID = "repurposed_structures";
+  private static final String RS_STRONGHOLD_ID = "stronghold_stonebrick";
+  private static final String RS_NETHER_STRONGHOLD_ID = "stronghold_nether";
+  private static final ResourceLocation RS_RESOURCE_LOCATION = new ResourceLocation(RS_MODID, RS_STRONGHOLD_ID);
```

Comment:
oh nice, thats easier than i expected. means we dont have to import any of the mods code